File: solutions/python/splitwise/group.py
import uuid
from typing import List, TYPE_CHECKING

if TYPE_CHECKING:
    from user import User

class Group:
    def __init__(self, name: str, members: List['User']):
        self._id = str(uuid.uuid4())
        self._name = name
        self._members = members
    
    def get_id(self) -> str:
        return self._id
    
    def get_name(self) -> str:
        return self._name
    
    def get_members(self) -> List['User']:
        return self._members.copy()

File: solutions/java/src/splitwise/Group.java
package splitwise;

import java.util.Date;
import java.util.List;
import java.util.UUID;
import java.util.concurrent.CopyOnWriteArrayList;
import java.util.stream.Collectors;

public class Group {
    private final String groupId;
    private final String name;
    private final List<User> members;
    private final List<Expense> expenses;
    private final Date createdAt;

    public Group(String name, List<User> members) {
        this.groupId = UUID.randomUUID().toString();
        this.name = name;
        this.members = members;
        this.expenses = new CopyOnWriteArrayList<>();
        this.createdAt = new Date();
    }

    public String getGroupId() {
        return groupId;
    }

    public String getName() {
        return name;
    }

    public List<User> getMembers() {
        return members;
    }

    public List<String> getMemberIds() {
        return members.stream()
                .map(User::getUserId)
                .collect(Collectors.toList());
    }
}
